### PR TITLE
Graphql polling log engine berk

### PR DESCRIPTION
**Pull request by QuiteStochastic**
https://github.com/MinaProtocol/mina/pull/13333

### CODE REVIEW

File: src/lib/logger/logger.mli [hunk 1]
```diff
@@ -85,8 +85,12 @@ module Processor : sig
 
   val raw : ?log_level:Level.t -> unit -> t
 
-  val pretty :
-    log_level:Level.t -> config:Interpolator_lib.Interpolator.config -> t
+
+  val raw_structured_log_events : Structured_log_events.Set.t -> t
+
+  (* val pretty : log_level:Level.t -> config:Logproc_lib.Interpolator.config -> t *)
```
Comment (mrmr1993) [inline src/lib/logger/logger.mli:91]: Delete

File: src/lib/logger/logger.mli [hunk 2]
```diff
@@ -103,6 +107,24 @@ module Transport : sig
 
   val create : (module S with type t = 'transport_data) -> 'transport_data -> t
 
+  module File_system : sig
```
Comment (mrmr1993) [inline src/lib/logger/logger.mli:110]: Delete

File: src/lib/logger/native/logger.ml [hunk 3]
```diff
@@ -228,6 +242,89 @@ module Transport = struct
   end
 
   let stdout () = T ((module Stdout), Stdout.create ())
+
+  module File_system = struct
```
Comment (mrmr1993) [inline src/lib/logger/native/logger.ml:246]: Delete

File: src/lib/logger/logger.mli (hunk 2 above)
Comment (QuiteStochastic) [reply src/lib/logger/logger.mli:110]: didn't you write this? what was this for again

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by QuiteStochastic** [comment]
!ci-build-me

File: src/lib/logger/logger.mli [hunk 4]
```diff
@@ -85,6 +85,8 @@ module Processor : sig
 
   val raw : ?log_level:Level.t -> unit -> t
 
+  val raw_structured_log_events : Structured_log_events.Set.t -> t
```
Comment (mitschabaude) [inline src/lib/logger/logger.mli:88]: to fix the JS build, you need to add fake implementations of these new methods to `logger/fake/logger.ml`

**Review by mitschabaude — APPROVED** [review]
looks great, fixing the JS build failure should be easy, let me know if you need help!

**Comment by QuiteStochastic** [comment]
!ci-build-me

**Comment by QuiteStochastic** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!approve-for-mainnet

**Comment by mrmr1993** [comment]
!approved-for-mainnet

**Pull request merged**